src/ts/ContentMetadataBuilder.ts

```diff
@@ -83,20 +81,26 @@ export class ContentMetadataBuilder {
     this.contentMetadata.defaultResource = this.metadataOverrides.defaultResource || this.metadata.defaultResource;
     this.contentMetadata.streamUrl = this.metadataOverrides.streamUrl || this.metadata.streamUrl;
 
-    const convivaContentInfo = {} as ConvivaMetadata;
-    convivaContentInfo[Conviva.Constants.ASSET_NAME] = this.contentMetadata.assetName;
-    convivaContentInfo[Conviva.Constants.ENCODED_FRAMERATE] = this.contentMetadata.encodedFrameRate;
-    convivaContentInfo[Conviva.Constants.DURATION] = this.contentMetadata.duration;
-    convivaContentInfo[Conviva.Constants.DEFAULT_RESOURCE] = this.contentMetadata.defaultResource;
-    convivaContentInfo[Conviva.Constants.STREAM_URL] = this.contentMetadata.streamUrl;
-    convivaContentInfo[Conviva.Constants.IS_LIVE] = this.contentMetadata.streamType;
-    convivaContentInfo[Conviva.Constants.VIEWER_ID] = this.contentMetadata.viewerId || 'GET_VIEWER_ID_FROM_PLAYER';
-    convivaContentInfo[Conviva.Constants.PLAYER_NAME] =
-      this.contentMetadata.applicationName || 'GET_PLAYER_NAME_OR_TYPE';
-    convivaContentInfo[Conviva.Constants.FRAMEWORK_NAME] = this.contentMetadata.framework;
-    convivaContentInfo[Conviva.Constants.FRAMEWORK_VERSION] = this.contentMetadata.frameworkVersion;
+    const convivaContentInfo: ReservedContentMetadata = {
+      [Conviva.Constants.ASSET_NAME]: this.contentMetadata.assetName,
+      [Conviva.Constants.ENCODED_FRAMERATE]: this.contentMetadata.encodedFrameRate,
+      [Conviva.Constants.DURATION]: this.contentMetadata.duration,
+      [Conviva.Constants.DEFAULT_RESOURCE]: this.contentMetadata.defaultResource,
+      [Conviva.Constants.STREAM_URL]: this.contentMetadata.streamUrl,
+      [Conviva.Constants.IS_LIVE]: this.contentMetadata.streamType,
+      [Conviva.Constants.VIEWER_ID]: this.contentMetadata.viewerId || 'GET_VIEWER_ID_FROM_PLAYER',
+      [Conviva.Constants.PLAYER_NAME]: this.contentMetadata.applicationName || 'GET_PLAYER_NAME_OR_TYPE',
+    };
+    const customContentInfo: CustomContentMetadata = {
+      [Conviva.Constants.FRAMEWORK_NAME]: this.contentMetadata.framework,
+      [Conviva.Constants.FRAMEWORK_VERSION]: this.contentMetadata.frameworkVersion,
```

Comment:
These two are not present in `ReservedContentMetadata` props, so they should be custom data.

Reply:
I think it would make sense to deprecate/remove those two options in that case. Customers can set any custom data anyway. What do you think?

Reply:
Done in https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/105/commits/04a0c70fd2c0e4686975bbefd62ad16af267adc0, https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/105/commits/529eb4d0064f1118cfa21efa8e9ea2b75f9ab5a9.